### PR TITLE
Adds k8s native pattern component generation

### PR DESCRIPTION
**Pull request by Revolyssup**
demo
**Description**

This PR adds k8s native pattern component generation

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: internal/store/store.go [hunk 1]
```diff
@@ -46,6 +46,14 @@ func Set(key string, value interface{}) {
 
 	globalStore.store[key][hash] = value
 }
+func Delete(key string, value interface{}) {
```
Comment (leecalcote) [inline internal/store/store.go:49]: Odd to think that we didn't have a Delete function previously.

File: models/pattern/core/register.go [hunk 2]
```diff
@@ -233,3 +236,75 @@ func registerMesheryServerOAM(rootPath string, constructs []string, regFn func([
...
+func GetK8Components(config []byte, ctx string) (*manifests.Component, error) {
+	cli, err := kubernetes.New(config)
+	if err != nil {
+		return nil, err
+	}
+	req := cli.KubeClient.RESTClient().Get().RequestURI("/openapi/v2")
+	res := req.Do(context.Background())
+	content, err := res.Raw()
+	if err != nil {
+		return nil, err
```
Comment (leecalcote) [inline models/pattern/core/register.go:249]: Need MeshKit-style errors.

File: models/pattern/core/register.go [hunk 3]
```diff
@@ -233,3 +236,75 @@ func registerMesheryServerOAM(rootPath string, constructs []string, regFn func([
...
+	}
+	apiResources, err := getAPIRes(cli)
+	if err != nil {
+		return nil, err
```
Comment (leecalcote) [inline models/pattern/core/register.go:253]: Will need a probable cause and suggested remediation for these errors.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: models/pattern/core/register.go [hunk 4]
```diff
@@ -233,3 +236,75 @@ func registerMesheryServerOAM(rootPath string, constructs []string, regFn func([
...
+				workload.OAMDefinition.APIVersion,
+				workload.OAMDefinition.Kind,
+				workload.OAMDefinition.Name,
+			)
+			store.Delete(key, value)
+		}
+	}
+}
+
+func getAPIRes(cli *kubernetes.Client) ([]string, error) {
```
Comment (leecalcote) [inline models/pattern/core/register.go:298]: @Revolyssup please document what each function does by commenting above each function.

File: models/pattern/core/register.go [hunk 5]
```diff
@@ -233,3 +236,75 @@ func registerMesheryServerOAM(rootPath string, constructs []string, regFn func([
...
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return man, nil
+}
+
+func DeleteK8sWorkloads(ctx string) {
+	//Iterate through entire store
```
Comment (leecalcote) [inline models/pattern/core/register.go:277]: Is this an in-memory store only? Or is this store the sqlite database?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: models/pattern/core/register.go (hunk 5 above)
Comment (Revolyssup) [reply models/pattern/core/register.go:277]: in memory store

**Comment by utkarsh-pro** [comment]
@Revolyssup, I missed this getting merged. Right now we are capturing everything about a component. Some of them are not required. Consider doing something like this
`kubeopenapi-jsonschema --location ./k8s.json -f '$.definitions' -t json --o-filter '$[0][?(@["x-kubernetes-group-version-kind"][0].kind=="Deployment")].properties.spec' --resolve "$"
`

**Comment by Revolyssup** [comment]
> @Revolyssup, I missed this getting merged. Right now we are capturing everything about a component. Some of them are not required. Consider doing something like this `kubeopenapi-jsonschema --location ./k8s.json -f '$.definitions' -t json --o-filter '$[0][?(@["x-kubernetes-group-version-kind"][0].kind=="Deployment")].properties.spec' --resolve "$" `

make the release  and I will create the PR
@utkarsh-pro  Does it resolve by default or the flag is necessary?
If necessary, I will have to add a field in meshkit and that means another meshkit release

**Comment by utkarsh-pro** [comment]
@Revolyssup refer description of https://github.com/layer5io/kubeopenapi-jsonschema/pull/4

**Pull request merged**